Src/Base/AMReX_FabArrayBase.cpp

```diff
@@ -665,24 +671,35 @@ FabArrayBase::FB::define_fb(const FabArrayBase& fa)
     for (int i = 0; i < nlocal; ++i)
     {
 	const int ksnd = imap[i];
-	const Box& vbx = ba[ksnd];
-	
+	const Box& vbx    = ba[ksnd];
+	const Box& vbx_ng = ba_ng[ksnd];
+
 	for (auto pit=pshifts.cbegin(); pit!=pshifts.cend(); ++pit)
 	{
 	    ba.intersections(vbx+(*pit), isects, false, ng);
+	    if (multi_ghost) ba_ng.intersections(vbx_ng+(*pit), isects_ng, false, ng_ng);
```

Comment:
I don't think it's necessary to define `ba_ng` and do the intersections here.  And it's probably wrong.  Because the code below assumes  isects and isects_ng have the same size and the same `first` for each item.  Because `ba_ng` is a completely different BoxArray and the intersection uses unordered_map internally, the order of the items might be different even though the size will be the same.  I think in the code below you can simply intersect the send box grown by one with the receive box grown by ng.

Reply:
Good point, I didn't realize that the map was unordered. So at the very least we need to iterate through `isects_ng` to ensure the keys match. That would be straightforward to implement.

I don't understand your solution though - wouldn't the send and receive boxes be too large in that case?

Reply:
I don't think you need to use `ba_ng` to get `isects_ng` at all.  When you iterate through `isects`, you know which two boxes gives that element of `isects`.  So you can easily compute what the corresponding element of `isects_ng` would be without having `isects_ng`.  We could have a zoom meeting next week to discuss this, if you want.

Reply:
@WeiqunZhang Ah, I understand now, thanks. I have implemented the change and removed `isects_ng`.